### PR TITLE
Feature/zcs 3177

### PR DESCRIPTION
**Pull request by tmclane**
Add menu options for enabling/disabling strict server name enforcement during install.
Upgrades will have it disabled.

### CODE REVIEW

File: rpmconf/Upgrade/zmupgrade.pm [hunk 1]
```diff
@@ -2047,6 +2047,14 @@ sub upgrade872GA {
     return 0;
 }
 
+sub upgrade886GA {
+    my ($startBuild, $targetVersion, $targetBuild) = (@_);
+    if (main::isInstalled("zimbra-proxy")) {
+        main::setLdapServerConfig($hn, 'zimbraReverseProxyStrictServerNameEnabled', "FALSE");
```
Comment (fatkudu) [inline rpmconf/Upgrade/zmupgrade.pm]: Please can we have a comment explaining why the upgrade setting differs from the new install setting.

Comment (tmclane) [reply rpmconf/Upgrade/zmupgrade.pm]: Sure thing

Comment (tmclane) [reply rpmconf/Upgrade/zmupgrade.pm]: How does this look?

File: rpmconf/Upgrade/zmupgrade.pm [hunk 2]
```diff
@@ -2047,6 +2047,19 @@ sub upgrade872GA {
...
 
+sub upgrade886GA {
+    my ($startBuild, $targetVersion, $targetBuild) = (@_);
+    if (main::isInstalled("zimbra-proxy")) {
+        # In order to avoid breaking existing installations 'Strict Server Name Enforcement' capability
+        # will remain disabled during upgrades. The maintainer may enable it after the upgrade process
+        # has completed and after configuring the appropriate domain(s) with the relevant
+        # 'zimbraVirtualHostName' and 'zimbraVirtualIPAddress' entries. This maintains current
+        # behavior for all upgrades until the administrator decides to enable the strict name enforcement.
+        main::setLdapServerConfig($hn, 'zimbraReverseProxyStrictServerNameEnabled', "FALSE");
```
Comment (tmclane) [inline rpmconf/Upgrade/zmupgrade.pm]: Here's the comments/description for why this is set the way it is.

**Pull request merged**